rust_snuba/src/mutations/clickhouse.rs

```diff
@@ -170,6 +170,76 @@ fn format_query(table: &str, batch: &MutationBatch) -> Vec<Vec<u8>> {
     ]
 }
 
+struct ClickhouseTestClient {
+    url: String,
+    table: String,
+    client: Client,
+}
+
+impl ClickhouseTestClient {
+    pub async fn new(table: String) -> anyhow::Result<Self> {
+        // hardcoding local Clickhouse settings
+        let hostname = "127.0.0.1";
+        let http_port = 9000;
+        let url = format!("http://{hostname}:{http_port}");
+
+        let client = reqwest::Client::new();
+        let test_table = format!("{table}_test_local\n");
+
+        let body =
+            format!("CREATE TABLE IF NOT EXISTS {test_table} AS {table}_local\n").into_bytes();
+
+        // use client to create a new table in local CH
+        client.post(url.clone()).body(body).send().await?;
+
+        Ok(Self {
+            url: url,
+            table: test_table,
+            client: client,
+        })
+    }
+
+    // actually run a query in the table
+    pub async fn run_query(&self, queries: Vec<Vec<u8>>) -> anyhow::Result<()> {
+        // run the vec of queries against Clickhouse
+        for q in queries {
+            self.client
+                .post(&self.url)
+                .body(q)
+                .send()
+                .await?
+                .error_for_status()?;
+        }
+
+        Ok(())
+    }
+
+    pub async fn insert_data(&self, primary_key: PrimaryKey) -> anyhow::Result<()> {
+        let organization_id = primary_key.organization_id;
+        let _sort_timestamp = primary_key._sort_timestamp;
+        let trace_id = primary_key.trace_id;
+        let span_id = primary_key.span_id;
+
+        let table = &self.table;
+
+        let insert =
+        format!("INSERT INTO {table} (organization_id, _sort_timestamp, trace_id, span_id, sign) VALUES ({organization_id}, {_sort_timestamp}, \'{trace_id}\', {span_id}, 1)\n").into_bytes();
+
+        self.client.post(&self.url).body(insert).send().await?;
+
+        Ok(())
+    }
+
+    pub async fn drop_table(&self) -> anyhow::Result<()> {
```

Comment:
do `impl Drop for ClickhouseTestClient`, and inside `fn drop()`, run `tokio::block_on(self.drop_table())`. then this function does not need to be called explicitly from the test, the table will just disappear when the test is over

Reply:
I tried this and for some reason it's causing the test to time out (hang) altogether, i.e:

`test mutations::clickhouse::tests::test_simple_mutation has been running for over 60 seconds`

I also tried a different variation and ran into `Cannot start a runtime from within a runtime.` I can dig deeper into this to understand what would work, but it seems a manual drop could also do? Let me know if you find another approach that works -- for supporting an asynchronous drop function

Reply:
Ok let's do it manually for now, we can fix it later

Reply:
I agree, I'd like to fix it so that I don't have to manually call drop. I'll keep looking into this on the side and if I find a solution that works I'll fix it.